### ISSUE TITLE
fix(model): enforce sampled MTP PRNG endpoint invariants

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12346]
Parent: #12235
Follow-on to closed #9996

## Classification
- **Centrality:** Core
- **Worker tier:** Astra mathematical contract/review; Sol implementation
- **Write set:** `internal/model/qwen38_mtp_sampled.go`, `internal/model/qwen38_mtp_sampled_test.go`

## For / Problem / Today / Better because / Witness
- **For:** native sampled MTP callers with target and draft distributions of different support.
- **Problem:** acceptance must implement `u < alpha` with generated `u in [0,1)`.
- **Today:** `randVal <= alpha` accepts an impossible draft when both equal zero. Separately, converting uint32 to float32 before scaling can round the PRNG output to 1.0.
- **Better because:** alpha=0 always rejects, alpha=1 always accepts, and the residual correction path remains distributionally valid.
- **Witness:** `TestQwen38MTPSampledAcceptanceEndpoints`.

## Acceptance
- Deterministically exercise the zero sample with alpha=0 and require residual replacement.
- Exercise the maximal uint32 endpoint and require `0 <= u && u < 1` plus alpha=1 acceptance.
- Review the predicate and fixed-width conversion as one coupled change.

## Witness command
`go test ./internal/model -run '^TestQwen38MTPSampledAcceptanceEndpoints$' -count=1`

## Non-goals
Full statistical calibration, sampler expansion, or PRNG algorithm replacement.

**Comment by anthony-chaudhary** [comment]
Claimed the bounded two-file leaf. The coupled fix will use the upper 24 PCG32 bits for an exactly representable float32 grid in [0,1), and strict u < alpha acceptance. One deterministic regression test forces PCG zero and maximal uint32 output, exercising residual replacement at alpha=0 and acceptance at alpha=1. No PRNG replacement or sampler expansion.

**Comment by anthony-chaudhary** [comment]
Implementation is prepared in the two assigned sampled-MTP files; issue remains open pending guarded landing after #12383.

Witness: go test ./internal/model -run "^TestQwen38MTPSampledAcceptanceEndpoints$" -count=1 failed before the fix with both zero/alpha=0 falsely accepted and maximal uint32 converted to u=1; the same command passed after the coupled conversion/predicate fix.

Additional green checks: go test ./internal/model -run "^TestQwen38MTP.*Sampled" -count=1; go vet ./internal/model; isolated fak validate selected sampled-MTP tests. Full package execution was killed with exit 137 and is not a passing receipt. Overall isolated validation remains red from the unrelated committed containsStr duplication (#12383) and an unavailable Strix extra gate (exit 255), non-applicable to this host-only mathematical fix. No gate was weakened.

Commit/push receipt is pending. A five-minute same-task follow-up will recheck committed trunk and complete guarded explicit-path landing once #12383 clears.

**Comment by anthony-chaudhary** [comment]
Dependency unblocked: #12383 is closed and commit 726c52b64 is contained in origin/main. The duplicate cmd/fak containsStr helper no longer blocks isolated validation.

**Comment by anthony-chaudhary** [comment]
After #12383 landed, restored the identical two-file patch and reran the endpoint witness, sampled-MTP suite, model vet, and isolated sampled tests: all green. Guarded reconciliation completed at 7e0581d35.

The explicit-path fak commit then refused with COMMITTED_RED (exit 4; compiled=true; nothing committed or pushed). Its full model package run failed TestQwen35PrefillAndStepCPUContinuation with NaN prefill logits, TestQwen35LinearAttnBatchedResumesState with a one-bit mismatch, and TestQwen35LinearAttnBatchedMatchesScalar plus TestQwen35ChunkedBatchedMatchesScalar with batched/scalar bit mismatches. Failure provenance still needs an independent committed-baseline witness. The unavailable Strix check also remains, tracked as the non-applicable extra gate in #12382.

Landing attempts are stopped at the typed refusal as instructed. The two-file patch and refusal receipt are preserved; issue remains open until a sanctioned green landing can finish.

**Comment by anthony-chaudhary** [comment]
Unblocker #12383 landed on `origin/main` in 726c52b6491dd479ccdc3f123bedb3318234281b (`cmd/fak@r4066+g726c52b64`). The duplicate `containsStr` declaration is gone; focused cmd/fak vet, tests, and build are green. Please retry the guarded commit/validation for #12346.

**Comment by anthony-chaudhary** [comment]
#12038 diagnosis is complete, but there is no pushed retry revision yet. The three batched/scalar mismatches share one tiled-SiLU rounding defect and are green after the bounded two-file compute candidate. The NaN is independent and separately tracked as #12433; its one-field fixture candidate is also green. Together, the unchanged #12346 patch plus candidate base was validated by the literal `go test -count=1 ./internal/model ./internal/gateway` gate (`model 403.667s`, `gateway 88.551s`).

Do not resume the preserved two-file endpoint patch yet: guarded landing of #12433 refused before staging solely on the unavailable automatic Strix phase (#12382), and no bypass was used. Exact retry condition is a pushed revision containing both #12433 and the #12038 compute correction, with changed `internal/model@rev` and `internal/compute@rev`. At that revision run `fak validate --ref <pushed-sha> --mine internal/model/qwen38_mtp_sampled.go --mine internal/model/qwen38_mtp_sampled_test.go`, then use guarded explicit-path commit/push only if green. The preserved #12346 files were not modified.

**Comment by anthony-chaudhary** [comment]
The prospective-candidate landing repair is shipped on `origin/main` in [`5623cb8df7c7`](https://github.com/anthony-chaudhary/fak/commit/5623cb8df7c701a3d334b50e89ab24afe716b077) (`cmd/fak r4070+g5623cb8df`, `internal/workerworktree r82+g5623cb8df`).

This removes the #12431 landing-path defect, but it does **not** clear this issue's independent committed-baseline blockers. Do not retry until these Qwen35 failures are resolved or explicitly sanctioned: `TestQwen35PrefillAndStepCPUContinuation`, `TestQwen35LinearAttnBatchedResumesState`, `TestQwen35LinearAttnBatchedMatchesScalar`, and `TestQwen35ChunkedBatchedMatchesScalar`. The Strix exit-255 dependency remains tracked by #12382.

After those blockers clear, rebuild the shipped binary and retry the existing explicit-path commit:

```bash
GOBIN=/Users/anthony/Documents/GitHub/fak/_scratch/issue-12346/bin \
  go install github.com/anthony-chaudhary/fak/cmd/fak@5623cb8df7c701a3d334b50e89ab24afe716b077

cd /Users/anthony/Documents/GitHub/fak
GOCACHE=/Users/anthony/Documents/GitHub/fak/_scratch/issue-12346/gocache \
GOMAXPROCS=2 GOFLAGS=-p=2 \
_scratch/issue-12346/bin/fak commit \
  --path internal/model/qwen38_mtp_sampled.go \
  --path internal/model/qwen38_mtp_sampled_test.go \
  -m 'fix(model): enforce sampled MTP endpoints #12346 (fak model)' \
  --build-check-timeout 10m --json \
  > _scratch/issue-12346/commit.json 2>&1
```

Run `fak sync push` only after that guarded commit returns green; do not use a raw merge, force push, or unsafe witness skip.

**Comment by anthony-chaudhary** [comment]
Retry precondition in addition to the baseline test blockers above: the selected checkout's `main` must contain `5623cb8df7c7`. Confirm with `git merge-base --is-ancestor 5623cb8df7c701a3d334b50e89ab24afe716b077 HEAD` plus `fak sync check --fetch`. The shared checkout is presently peer-dirty, so preserve its WIP and use only sanctioned `fak sync reconcile` convergence if that ancestry check is false; do not reset, raw-merge, or force-push it.
